Makefile

```diff
@@ -580,16 +580,16 @@ pretty/LaTeX_maps.hs: utils/words.pl utils/genItCorrections \
 
 ### clean up
 clean_stack:
-	rm -rf .stack-work
+	@$(RM) -rf .stack-work
 
 clean_genRules:
 	@$(RM) $(generated_rule_files) $(gendrifted_files) $(hs_clean_files)
 
-### removes all *.o, *.hi and *.p_o files in all subdirectories
+### removes all *.o, *.hi and *.p_o files in all subdirectories except for .stack-work
 o_clean:
-	@find . \( -path ./.stack-work -prune \
-		-name '*.o' -o -name '*.hi' -o -name '*.p_o' \
-        -o -name '*.dyn_hi' -o -name '*.dyn_o' \) -exec rm -f {} +
+	@find . \( -name '*.o' -o -name '*.hi' -o -name '*.p_o' \
+			   -o -name '*.dyn_hi' -o -name '*.dyn_o' \) \
+			-o -path ./.stack-work -prune -type f -exec rm -f {} +
```

Comment:
Ehmm, no - I doubt that it does, what it should ;-) - shortcut rules are in effect! I.e. if for `(A) || B` A evals to true, B gets not evaluated. So actually the expression above just considers everything not matching the patterns enclosed in () as rm candidate (subject for B). Unfortunately (if -path _pattern_ works as you say), the only candidate for -exec expression becomes the file ./.stack-work, so IMHO under normal conditions the find expressions always evaluates to false/doesn't remove anything.